src/controllers/expenses.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const expensesServices = require('../services/expennses');
+const userServices = require('../services/users');
+
+const getAll = (req, res) => {
+  const partsUrl = req.url.split('?');
+  const urlParams = new URLSearchParams(partsUrl[1]);
+
+  const userId = urlParams.get('userId');
+  const category = urlParams.get('category');
+  const from = urlParams.get('from');
+  const to = urlParams.get('to');
+
+  let expenses = expensesServices.getAllExpenses();
+
+  if (userId) {
+    expenses = expenses.filter(el => el.userId === +userId);
+  }
+
+  if (category) {
+    expenses = expenses.filter(el => el.category === category);
+  }
+
+  if (from && to) {
+    expenses = expenses.filter(el => el.spentAt > from && el.spentAt < to);
+  }
+
+  res.send(expenses);
```

Comment:
in my opinion controllers shouldn't have as much business logic they should be as simple as it can be. receive a request call a service send a response. So better move it to the services

Reply:
Fix it in all your controllers